packages/rs-dpp/src/data_contract/document_type/v0/random_document_type.rs

```diff
@@ -218,12 +218,42 @@ impl DocumentTypeV0 {
             .cloned()
             .collect_vec();
 
+        // Generate indices
         let mut indices = Vec::with_capacity(index_count as usize);
+        let mut existing_combinations = HashSet::new();
 
         for _ in 0..index_count {
-            match Index::random(&ten_field_names, &indices, rng) {
-                Ok(index) => indices.push(index),
-                Err(_) => break,
+            let mut attempt = 0;
+            let max_attempts = 10;
+            let mut index_created = false;
+
+            while attempt < max_attempts && !index_created {
+                attempt += 1;
+                match Index::random(&ten_field_names, &indices, rng) {
+                    Ok(index) => {
+                        // Generate a key to represent the combination of properties in this index
+                        let combination_key = index
+                            .properties
+                            .iter()
+                            .map(|p| p.name.clone())
+                            .collect::<Vec<_>>()
+                            .join(",");
+
+                        // Check if this combination has already been used
+                        if existing_combinations.insert(combination_key) {
+                            // This is a new combination, so we can add the index
+                            indices.push(index);
+                            index_created = true;
+                        }
+                    }
+                    Err(_) => break,
+                }
+            }
+
+            if !index_created {
+                // Unable to create a unique index after several attempts
+                // You might want to handle this case, e.g., by logging a warning or adjusting the logic
+                break;
```

Comment:
This isn't needed... all you need to do instead is add the random index to indices. This is because ```Index::random(&ten_field_names, &indices, rng) {``` already checks the newly created index isn't in indices.

Reply:
Actually I was getting a lot of DuplicateIndexError until I added this

Reply:
Read my comment.

Reply:
Resolution: apparently Index::random() was not properly ensuring uniqueness, so we fixed that and removed the checking from random_document_types()